tensorflow/lite/interpreter_builder.cc

```diff
@@ -168,9 +168,10 @@ TFLITE_ATTRIBUTE_WEAK Interpreter::TfLiteDelegatePtr AcquireFlexDelegate() {
   // important that on Android 4.4 we *don't* call SharedLibrary::GetSymbol
   // unless the symbol is sure to exist.
 #if !defined(__ANDROID__)
+
   auto acquire_flex_delegate_func =
       reinterpret_cast<Interpreter::TfLiteDelegatePtr (*)()>(
-          SharedLibrary::GetSymbol("TF_AcquireFlexDelegate"));
+          SharedLibrary::GetSymbol2(SharedLibrary::LoadLibrary(dll_path_string.c_str()), "TF_AcquireFlexDelegate"));
```

Comment:
Every call to the Windows LoadLibrary API call (which I assume SharedLibrary::LoadLibrary makes) should be followed by a call to ::FreeLibrary when you are done using the library.

When a process exits, all libraries that have not been explicitly unloaded (via FreeLibrary) will be unloaded by the OS. This is okay, but can sometimes lead to memory leak detection utilities to falsely report a memory leak.

LoadLibrary and FreeLibrary use reference counting, if I recall correctly, so that if you call LoadLibrary twice in a row for the same DLL, FreeLibrary would also have to be called twice.

I don't see anywhere in this PR where you call FreeLibrary, which is why I raise this.

Should this call to LoadLibrary be made at a higher scope? Perhaps it would make sense to pass the library handle as an arg to this method rather than passing the filename of the DLL? Then you could have better control ofthe scope of the lifetime of the loaded DLL.

Reply:
@efieleke-tausight I agree, updated the code.